File: backend/spring/pulse/src/main/java/dev/thilanka/netrics/dto/WorstCellsWithLatestDto.java
package dev.thilanka.netrics.dto;

public record WorstCellsWithLatestDto(
        Long id,
        String cellName,
        String kpiName,
        String kpiLabel,
        String unit,
        Double value,
        Double previousValue,
        Double difference,
        boolean improved,
        Double latestValue,
        Double latestDifference,
        boolean latestImproved
) {
}

File: frontend/netrics/src/app/models/pulse/WorstCellsWithLatestDto.ts
export interface WorstCellsWithLatestDto {
  id: number;
  cellName:string;
  kpiName: string;
  kpiLabel:string;
  unit:string;
  value:number;
  previousValue:number;
  difference:number;
  improved:boolean;
  latestValue:number;
  latestDifference:number;
  latestImproved:boolean;
}
